evalml/pipelines/component_graph.py

```diff
@@ -0,0 +1,310 @@
+import networkx as nx
+import pandas as pd
+from networkx.algorithms.dag import topological_sort
+
+from evalml.pipelines.components import ComponentBase, Estimator, Transformer
+from evalml.pipelines.components.utils import handle_component_class
+from evalml.utils import import_or_raise
+
+
+class ComponentGraph:
+    def __init__(self, component_dict=None, random_state=0):
+        """ Initializes a component graph for a pipeline as a DAG.Example:
+
+        Example:
+            >>> component_dict = {'imputer': ['Imputer'], 'ohe': ['One Hot Encoder', 'imputer.x'], 'estimator_1': ['Random Forest Classifier', 'ohe.x'], 'estimator_2': ['Decision Tree Classifier', 'ohe.x'], 'final': ['Logistic Regression Classifier', 'estimator_1', 'estimator_2']}
+            >>> component_graph = ComponentGraph(component_dict)
+           """
+        self.component_dict = component_dict or {}
+        for key, value in self.component_dict.items():
+            if not isinstance(value, list):
+                raise ValueError('All component information should be passed in as a list')
+        self._compute_order = []
+        self._recompute_order()
+        self.random_state = random_state
+
+    @classmethod
+    def from_list(cls, component_list, random_state=0):
+        """Constructs a linear ComponentGraph from a given list, where each component in the list feeds its output to the next component
+
+        Arguments:
+            component_list (list): String names or ComponentBase subclasses in
+                                   an order that represents a valid linear graph
+        """
+        component_dict = {}
+        for idx, component in enumerate(component_list):
+            component_class = handle_component_class(component)
+            component_name = component_class.name
+
+            component_dict[component_name] = [component_class]
+            if idx != 0:
+                component_dict[component_name].append(f"{handle_component_class(component_list[idx-1]).name}.x")
```

Comment:
@eccabay to close off [the thread](https://github.com/alteryx/evalml/pull/1415#discussion_r522152500) from an outdated version of this code: looks good! Could you please mention in the docstring that this method will pass along the "x" and not the "y"?